src/tests/unit/common/fail-test-on-error-logger.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { inspect } from 'util';
+import { Logger } from 'common/logging/logger';
+
+export const failTestOnErrorLogger: Logger = {
+    log: () => {
+        // intentionally ignored
+    },
+    error: (message?: string, ...optionalParams: any[]) => {
+        fail(`FailTestOnErrorLogger.error invoked with "${message}" ${inspect(optionalParams)}`);
```

Comment:
I looked for this and found https://github.com/facebook/jest/issues/2129 mentioning it's a feature of jasmine, but I'm not sure if I'm missing something related to jest-circus that will help us use it. I added `fail` locally to a test and it failed because fail() was not defined. Is there an API I'm missing, or should we do this a different way?

Reply:
Huh; I had used this based on this from `@types/jest/index.d.ts`:

```ts
/**
 * Fails a test when called within one.
 */
declare function fail(error?: any): never;
```

Based on that thread, I'll switch it to something based on `expect()`. Great catch!